metrics/integration_test.go

```diff
@@ -101,6 +102,47 @@ var _ = Describe("Metrics Integration", Ordered, func() {
 			Expect(testutil.CollectAndCompare(SnapshotCreatedToPipelineRunStartedSeconds, strings.NewReader(readerData))).To(Succeed())
 		})
 	})
+
+	Context("When RegisterPipelineRunWithEphemeralEnvStarted is called", func() {
+		// As we need to share metrics within the Context, we need to use "per Context" '(Before|After)All'
+		BeforeAll(func() {
+			// Mocking metrics to be able to resent data with each tests. Otherwise, we would have to take previous tests into account.
+			//
+			// 'Help' can't be overridden due to 'https://github.com/prometheus/client_golang/blob/83d56b1144a0c2eb10d399e7abbae3333bebc463/prometheus/registry.go#L314'
+			SnapshotCreatedToPipelineRunWithEphemeralEnvStartedSeconds = prometheus.NewHistogram(
+				prometheus.HistogramOpts{
+					Name:    "snapshot_created_to_pipelinerun_started_seconds",
+					Help:    "Time duration from the moment the snapshot resource was created till a integration pipelineRun is started",
+					Buckets: []float64{1, 5, 10, 30},
+				},
+			)
+		})
+
+		AfterAll(func() {
+			metrics.Registry.Unregister(SnapshotCreatedToPipelineRunWithEphemeralEnvStartedSeconds)
+		})
+
+		// Input seconds for duration of operations less or equal to the following buckets of 1, 5, 10 and 30 seconds
+		inputSeconds := []float64{1, 3, 8, 15}
+		elapsedSeconds := 0.0
+
+		It("registers a new observation for 'snapshot_created_to_pipelinerun_with_ephemeral_env_started_seconds' with the elapsed time from the moment"+
+			"the snapshot is created to first integration pipelineRun is started.", func() {
+			creationTime := metav1.Time{}
+			for _, seconds := range inputSeconds {
+				RegisterNewSnapshot()
```

Comment:
Why do we need this function here?

Reply:
Totally not resolved. Why we need to keep it commented?

Also I'd appreciate better communication than just silent resolving of comments.

Reply:
Sorry about that, most of the issues you brought up were things that I copied from another test and forgot to delete.  I thought I had deleted that line since it doesn't seem to be necessary but it looks like I had just commented it out.  Should be all set now.